### ISSUE TITLE
QLinearAdd Op Request

### ISSUE THREAD

**Issue by cjvolzka** [onnx/onnx#5895]
# QLinearAdd Operator Request

### Describe the operator
An Add Operator for quantized data. It supports zero_point and scale input tensor for the Add input (A and B) and output (C) tensors:
This Op exists in ONNXRuntime time but not in the ONNX standard operators
https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.QLinearAdd

Several int8.onnx models in the onnx model zoo validated directory user this Op. Pretty much all of them but one which have QlinearMatMul also have QLinearAdd.

Onnx-mlir would like to support these models but we only support official ONNX Operators. 

### Can this operator be constructed using existing onnx operators?
Unsure. 

### Is this operator used by any model currently? Which one?
Several offhand I found in the onnx model zoo:
[bvlcalexnet-12-int8](https://github.com/onnx/models/blob/main/validated/vision/classification/alexnet/model/bvlcalexnet-12-int8.onnx)
[mnist-12-int8](https://github.com/onnx/models/blob/main/validated/vision/classification/mnist/model/mnist-12-int8.onnx)
[vgg16-12-int8](https://github.com/onnx/models/blob/main/validated/vision/classification/vgg/model/vgg16-12-int8.onnx)

### Are you willing to contribute it? (Y/N)
N

### Notes
ONNX already has QLinearMatMul and QLinearConv from these models but appears to be missing QLinearAdd.

**Comment by justinchuby** [comment]
Can this be represented using the Dequantize-Add-Quantize pattern?

**Comment by cjvolzka** [comment]
No. Dequantizing would turn them back into floats. So you'd loose both the memory savings and performance improvements of integer math on the operation. You'd also have to recalculate scales to requantize which would add time. Also using the QLinear* Ops are for models that were quantized at training time. So dequantizing and requantizing would incur accuracy hits hat would harm the point of the "quantization aware" training of the the original model.

Also from what I've seen, when you have QLinear* Ops you start with a QuantizeLinear. Do a series of QLinear* Ops on the values and then afterward there's a DequantizeLinear at the end. Everything between the QuantizeLinear and DequantizeLinear should stay quantized and use scales and offsets set at training time.
